src/minimega/vnc.go

```diff
@@ -61,37 +60,21 @@ func init() {
 
 // NewVNCClient creates a new VNC client. NewVNCClient is only called via the
 // cli so we can assume that cmdLock is held.
-func NewVNCClient(host, idOrName string) (*vncClient, error) {
-	// Resolve localhost to the actual hostname
-	if host == Localhost {
-		host = hostname
-	}
-
+// This is sent via wrapVMTargetCLI so we assume the command will always be
+// delivered to the correct host
+func NewVNCClient(name string) (*vncClient, error) {
 	var vm *KvmVM
-	if host == hostname {
-		vm, _ = vms.FindKvmVM(idOrName)
-	} else {
-		// LOCK: This is only invoked via the CLI so we already hold cmdLock
-		// (can call hostVMs instead of HostVMs). Since we're using not using
-		// the vms global, we don't need to acquire the vmLock (can call findVM
-		// instead of FindVM).
-
-		// TODO(fritz): should this be namespace aware? If someone sets
-		// a namespace on the cli and then someone on the web interface
-		// attempts to connect and this is checking namespaces then it
-		// will fail right?
-		vm, _ = hostVMs(host).findKvmVM(idOrName)
-	}
+	vm, _ = vms.FindKvmVM(name)
 
 	if vm == nil {
-		return nil, vmNotFound(host + ":" + idOrName)
+		return nil, vmNotFound(name)
 	}
 
-	rhost := fmt.Sprintf("%v:%v", host, vm.VNCPort)
+	rhost := fmt.Sprintf("%v:%v", name, vm.VNCPort)
 
 	c := &vncClient{
 		Rhost: rhost,
-		Host:  host,
+		Host:  hostname,
```

Comment:
Might be worth adding a reference to the VM to the `vncClient` struct. Then you don't need to store Name or ID.